File: .config/rofi/notfi/clear.js
#!/usr/bin/node

const fs = require('fs');

const path = process.argv[2];

const [ timestamp, summary, title ] = process.argv[3].split('\\n');

const content = fs.readFileSync(path, { encoding: 'utf-8' }).split('\n');

console.log(content);

const target = content.findIndex((s, i) =>
{
  if (s === timestamp && content[i + 3] === summary && content[i + 4] === title)
    return true;
});

if (target <= -1)
  return;

content.splice(target, 6);

fs.writeFileSync(path, content.join('\n'));

File: .config/rofi/notfi/clear.py
#!/usr/bin/python

import sys

args = sys.argv[1].split('\\n');

timestamp = args[0]
summary = args[1]
name = args[2]

with open('/tmp/dunstlog', 'r') as file:
  lines = file.read().splitlines()

with open('/tmp/dunstlog', 'w') as file:
  index = -1

  for i, s in enumerate(lines):
    if s == timestamp and lines[i+3] == summary and lines[i+4] == name:
      index = i
      break
    elif s == timestamp and lines[i+4] == summary and lines[i+5] == name:
      index = i
      break

  if index > -1:
    del lines[index:index+6]
  
  file.write('\n'.join(lines) + '\n')